app/pb/blocking.proto

```diff
@@ -0,0 +1,29 @@
+syntax = "proto3";
+
+package org.couchers.blocking;
+
+import "google/protobuf/empty.proto";
+
+service Blocking {
+  // This is a secure service: a user needs to be authenticated and not jailed to call functions here, refer to auth.proto and jail.proto
+
+  rpc BlockUser(BlockUserReq) returns (google.protobuf.Empty);
+
+  rpc UnblockUser(UnblockUserReq) returns(google.protobuf.Empty);
+
+  rpc GetBlockedUsers(google.protobuf.Empty) returns (GetBlockedUsersRes) {
+    //Returns all users blocked by a given account
+  }
+}
+
+message BlockUserReq {
+  int64 user_id = 1;
+}
+
+message UnblockUserReq {
+  int64 user_id = 1;
```

Comment:
@kalvdans: I think these `user_id`s should be changed to `string username`.

There's no way for the requester to get the user_id because `GetUser` will return not found for blocked users.

Also if user A blocks user B, then user B might also want to block user A, so it's not guaranteed that a the blocker can find the user id either

Reply:
If I block you, I should still be able to GetUser you IMHO. But indeed if we have a mutual blockage I guess I have to call you and ask which user_id number you have.

Reply:
I think it's better to just use username here.

Regarding getting the user with GetUser; there should maybe be an option to get some subset of information. But I don't think it's very good that the blocker has the power to see the other user but the other user can't see the blocker, could be abused.

Reply:
I guess one can argue either way. Symmetrical blocking is conceptually simpler indeed.

OK, please go with usernames here and in GetBlockedUsers